artifacts/reader.py

```diff
@@ -72,6 +72,10 @@ def _ReadArtifactDefinition(self, yaml_definition):
     if not yaml_definition:
       raise errors.FormatError(u'Missing YAML definition.')
 
+    different_keys = set(yaml_definition) - definitions.TOP_LEVEL_KEYS
+    if different_keys:
+      raise errors.FormatError(u'Bad keys: {0!s}'.format(different_keys))
```

Comment:
> The intention here is to catch mistyped optional keys like "urps" instead of "urls".

I see, it was not clear to me from the description in https://github.com/google/grr/issues/277 that it was caused by a typo.

> {0:s} vs. {0!s}

I meant to use {0!s} in the example.

> Again, my preference is not to have the explicit conversion for strings

If we want to change it, that fine with me if we do it for the whole codebase at once. I would opt that we don't mix styles.

> I'll change the error text here if you like

Then I opt:
```
raise errors.FormatError(u'Unsupported keys: {0!s}'.format(different_keys))
```

And remove the {0:s} type notation in a follow up CL.

> This also isn't a format change so I don't think there's any need to discuss with others.

I meant this as part of the error message, when the error is raised to indicate changes might need to be added to the format.